File: backend/api/routes/root.js
'use strict'
module.exports = async function (fastify, opts) {
  fastify.get('/', async function (request, reply) {
    return { root: true }
  })

  // 3. Deep Health Probes (Enterprise Readiness)
  fastify.get('/health', async function (request, reply) {
    try {
      const db = require('../controllers/db')
      // Actually execute a Knex query to guarantee the connection pool is healthy
      await db.knex.raw('SELECT 1')
      return reply.code(200).send({ status: 'ok', db: 'connected' })
    } catch (err) {
      fastify.log.error({ err }, 'Deep health check failed - DB disconnected')
      return reply.code(503).send({ status: 'error', db: 'disconnected' })
    }
  })
}

File: backend/api_fastapi/src/csearch_api/routes/root.py
from __future__ import annotations

from fastapi import APIRouter, HTTPException, Request

router = APIRouter()


@router.get("/")
async def root():
    return {"root": True}


@router.get("/health")
async def health(request: Request):
    try:
        await request.app.state.db.fetchval("SELECT 1")
        return {"status": "ok", "db": "connected"}
    except Exception:
        raise HTTPException(status_code=503, detail={"status": "error", "db": "disconnected"})

